File: semester_4_(spring-23)/data_structures_and_algorithms/assignments/assignment1/bubbleSort.cpp
/*
Q2:
Write a C++ program to sort a list of elements using the bubble sort algorithm.
Note: Bubble Sort works by swapping adjacent elements if they are in the wrong order.
*/
#include <iostream>
using namespace std;

void swapBoth(int &x, int &y){
    int temp = x;
    x = y;
    y = temp;
}

void displayArray(int* arr, int &ofLength){
    for(int i = 0; i < ofLength; i++)
        cout<<arr[i]<<" ";
    cout<<endl;
}

int* bubbleSort(int* arr, int &arrSize){
    for (int i = 0; i < arrSize; i++)
        for(int j = 0; j < arrSize-1; j++)
            if(arr[j] > arr[j + 1])
                swapBoth(arr[j], arr[j + 1]);

    return arr;
}

int main(){
    int arrSize = 10;
    int arr[arrSize] = {5, 1, 4, 2, 8, 7 ,9, 10, 87, -3};

    cout<<"Unsorted Array: "<<endl;
    displayArray(arr, arrSize);

    int *sortedArr = bubbleSort(arr, arrSize);

    cout<<"Sorted Array: "<<endl;
    displayArray(sortedArr, arrSize);

    return 0;
}
//Aimal Khan.

File: semester_4_(spring-23)/data_structures_and_algorithms_lab/lab_reports/lab2/bubbleSort.py
def bubbleSort (array):
    for step in range(0, len(array)):
        for i in range(0, len(array)-1):
            if(array[i] > array[i + 1]):
                array[i + 1], array[i] = array[i], array[i + 1]
    
    return array

print("Unsorted list:")
data = [2, -7, 8 , 2, -3, 5, 0 , 1]
print(data)

print("Sorted list:")
sortedData = bubbleSort(data)

print(sortedData)